rir/src/compiler/analysis/last_env.h

```diff
@@ -0,0 +1,33 @@
+#ifndef PIR_LAST_ENV_H
+#define PIR_LAST_ENV_H
+
+#include "abstract_value.h"
+#include "generic_static_analysis.h"
+
+namespace rir {
+namespace pir {
+
+class LastEnv : public StaticAnalysis<AbstractUnique<Value>> {
+  public:
+    LastEnv(ClosureVersion* cls, Code* code, LogStream& log)
+        : StaticAnalysis("Last Env", cls, code, log) {}
+
+    AbstractResult apply(AbstractUnique<Value>& state,
+                         Instruction* i) const override {
+        if (i->hasEnv() && !MkEnv::Cast(i) && i->env() != state.get()) {
```

Comment:
I am still confused by `MkEnv` vs. `LdFunctionEnv`... This looks like exactly the same condition as the p2r compiler `explicitEnvValue`. Why is one explicit and the other not?
Or, shouldn't this test also check the other one?

Reply:
This is about the arguments. For almost all instruction an environment argument means, that the env should be set using `set_env_`. Except `mk_env_`, this one expects the parent env as a normal argument on the stack and not as the value of `*env`.